docs/user-guides/acme-dns-validation.md

```diff
@@ -0,0 +1,94 @@
+
+The first stage of ACME is for the client to register with the ACME server. This phase includes generating an asymmetric key pair which is then associated with the email address specified in the `Issuer`. Depending on the CA, this could be used to send expiry notices when your certificates are coming up for renewal. The generated private key is stored in a `Secret` called `letsencrypt-staging`.
+
+The `dns01` stanza contains a list of `dns-01` providers that can be used to solve DNS challenges. A list of currently supported providers can be found [here][5]
+
+Once we have created the above `Issuer` we can use it to obtain a certificate.
+
+```yaml
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Certificate
+metadata:
+  name: nginx-k8s-io
+  namespace: default
+spec:
+  secretName: nginx-k8s-io-tls
+  issuerRef:
+    name: letsencrypt-staging
+  commonName: nginx.k8s.io
+  dnsNames:
+  - nginx2.k8s.io
+  - nginx3.k8s.io
+  acme:
+    config:
+    - dns01:
+        provider: clouddns
+      domains:
+      - nginx.k8s.io
+    - dns01:
+        provider: cloudflare
+      domains:
+      - nginx2.k8s.io
+      - nginx3.k8s.io
+```
+
+The `Certificate` resource describes our desired certificate and the possible methods that can be used to obtain it. You can learn more about the `Certificate` resource [here][7]. If the certificate is obtained successfully the resulting key pair will be stored in a secret called `nginx-k8s-io-tls` in the same namespace as the `Certificate`. The certificate will have a common name of `nginx.k8s.io` and the [Subject Alternative Names][6] (SANs) will be `nginx.k8s.io`, `nginx2.k8s.io` and `nginx3.k8s.io`.
+
+In our `Certficate` we have referenced the `letsencrypt-staging` `Issuer` above. The `Issuer` must be in the same namespace as the `Certficate`. If you want to reference a `ClusterIssuer`, which is a cluster-scoped version of an `Issuer`, you must add `kind: ClusterIssuer` to the `issuerRef` stanza.
+
+The `acme` stanza defines the configuration for our ACME challenges. Here we have defined the configuration for our DNS challenges which will be used to validate our domains. For each domain mentioned in a `dns01` stanza, `cert-manager` will use the provider's credentials from the referenced `Issuer` to create a TXT record called `_acme-challenge`. This record will then be verified by the ACME server in order to issue the certificate. Note that it is your responsibility to ensure the provider is authoritative for your domain. 
+
+Once our certificate has been obtained, `cert-manager` will keep checking its validity and attempt to renew it if it gets close to expiry. `cert-manager` considers certificates to be close to expiry when the 'Not After' field on the certificate is less than the current time plus 30 days.
+
+Once we delete our `Certificate` resource, any `cert-manager` affected records will be cleaned up.
```

Comment:
This is not the case - the records will be cleaned up after a validation is successful: https://github.com/jetstack-experimental/cert-manager/blob/master/pkg/issuer/acme/dns/dns.go#L95